src/bp-members/bp-members-filters.php

```diff
@@ -830,39 +830,55 @@ function bb_repair_member_profile_links_callback( $is_background = false, $paged
 		return;
 	}
 
+	global $wpdb;
+
+	// Define default values.
 	$per_page = 50;
+	$offset   = 0;
+
+	if ( ! $is_background ) {
 
-	if ( $is_background ) {
-		$offset = ( ( $paged - 1 ) * $per_page );
-	} else {
 		// phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash
 		$offset = isset( $_POST['offset'] ) ? (int) ( $_POST['offset'] ) : 0;
-	}
 
-	$args = array(
-		'fields' => 'ID',
-		'number' => $per_page,
-		'offset' => $offset,
-	);
+		// Set limit while repair the member slug.
+		$user_ids = $wpdb->get_col(
+			$wpdb->prepare(
+				"SELECT user_id FROM `{$wpdb->prefix}usermeta` WHERE LENGTH(meta_value) = %d AND `meta_key` = %s ORDER BY user_id ASC LIMIT %d, %d",
+				40,
+				'bb_profile_slug',
+				0,
+				$per_page
+			)
+		);
```

Comment:
Please update the query, this will generate the user profile slug for those users which have 40 character length, but it will not generate for those users where the profile slug was not generated.